copis/gui/panels/machine_toolbar.py

```diff
@@ -16,72 +16,77 @@
 """MachineToolbar class."""
 
 import logging
-
+import threading
 import wx
 import wx.lib.agw.aui as aui
+
 from copis.enums import ToolIds
 from copis.gui.machine_settings_dialog import MachineSettingsDialog
 from copis.gui.wxutils import create_scaled_bitmap, set_dialog
-from copis.coms.serial_controller import SerialController
 
 
 class MachineToolbar(aui.AuiToolBar):
     """Manage AUI toolbar panel."""
 
-    def __init__(self, parent, *args, **kwargs) -> None:
+    def __init__(self, parent, console = None) -> None:
         """Inits MachineToolbar with constructors."""
         super().__init__(parent, style=wx.BORDER_DEFAULT, agwStyle=
             aui.AUI_TB_PLAIN_BACKGROUND|aui.AUI_TB_OVERFLOW)
-        self.parent = parent
-        self.core = self.parent.core
-
-        self.serial_controller = None
+        self._parent = parent
+        self._core = self._parent.core
+        self._serial = self._core.serial
+        self._console = console
 
         self.port_cb = None
         self.baud_cb = None
 
-        self.init_controller()
         self.init_toolbar()
         self.update_ports()
 
         self.Bind(wx.EVT_TOOL, self.on_tool_selected)
 
-    def init_controller(self) -> None:
-        """Initialize serial controller object."""
-        if self.serial_controller is not None:
-            return
-
-        self.serial_controller = SerialController()
-
     def init_toolbar(self) -> None:
         """Initialize and populate toolbar.
 
         Icons taken from https://material.io/resources/icons/?style=baseline.
         """
-        # add port, baud comboboxes
+        # Add port, baud comboboxes.
```

Comment:
Inline comments don't need to be ended w/ a period

Reply:
I like terminate my sentences.